olga/charts/views.py

```diff
@@ -2,78 +2,59 @@
-        """
-        Passes graph data to frontend.
-        """
-
+    try:
         first_datetime_of_update_data = InstallationStatistics.objects.first().data_created_datetime
         last_datetime_of_update_data = InstallationStatistics.objects.last().data_created_datetime
+    except AttributeError:
+        first_datetime_of_update_data = datetime.datetime.now
+        last_datetime_of_update_data = datetime.datetime.now
 
-        worlds_students_per_country = InstallationStatistics.worlds_students_per_country_statistics()
-
-        datamap_format_countries_list = []
-        tabular_format_countries_list = []
-
-        all_active_students = sum(worlds_students_per_country.itervalues())
-
-        for country, count in worlds_students_per_country.iteritems():
-            student_amount_percentage = format(count/all_active_students*100, '.2f')
+    return first_datetime_of_update_data, last_datetime_of_update_data
 
-            # Too small percentage doesn't show real numbers. More than two numbers after point is ugly.
-            if student_amount_percentage == '0.00':
-                student_amount_percentage = '~0'
 
-            if country != 'null':
-                # Make data to datamap visualization format.
-                datamap_format_countries_list.append([
-                    str(pycountry.countries.get(alpha_2=country).alpha_3), count
-                ])
-
-                # Make data to simple table visualization format.
-                tabular_format_countries_list.append((
-                    pycountry.countries.get(alpha_2=country).name, count, student_amount_percentage
-                ))
-
-            else:
-                # Create students without country amount.
-                tabular_format_countries_list.append(('Unset', count, student_amount_percentage))
+class MapView(View):
+    """
+    Display information on a world map and tabular view.
+    """
 
-        # Sort in descending order.
-        tabular_format_countries_list.sort(key=lambda row: row[1], reverse=True)
+    @staticmethod
+    def get_statistics_top_country(tabular_format_countries_list):
+        """
+        Get first country from tabular format country list.
 
-        # Workaround when there is no data for the given day.
+        List is sorted, first country is a top active students rank country.
+        """
         if not tabular_format_countries_list:
-            tabular_format_countries_list.append(('Unset', 0, 0))
+            return None
 
-            # Unset is not a country
-            countries_amount = 0
+        return tabular_format_countries_list[0][0]
 
-        else:
-            # Delete unset country point from list
-            countries_amount = len(tabular_format_countries_list) - 1
+    def get(self, request):
+        """
+        Pass graph data to frontend.
+        """
+        first_datetime_of_update_data, last_datetime_of_update_data = get_first_and_last_datetime_of_update_data()
```

Comment:
> Why somebody should search if these variables are internal to the function?

Somebody needs to know - what does variable mean.
If I see `first_datetime_of_update_data` I understand value.
If I see `first_dt` I do not.